src/vg/civcraft/mc/civmenu/TOSListener.java

```diff
@@ -66,13 +65,27 @@ public void run() {
 	public void playerMoveEvent(PlayerMoveEvent event) {
 		Player p = event.getPlayer();
 		if (!TOSManager.isTermPlayer(p, "CivMenu Agreement")) {
-			if(event.getTo().distance(locations.get(p.getUniqueId())) > plugin.GetConfig().get("terms.MovementRange").getInt()){
+			Location from = locations.get(p.getUniqueId());
+			if (from == null){
+				from = event.getFrom();
+				locations.put(p.getUniqueId(), from);
```

Comment:
Unregistered players not bound to any location should be caught by these changes. Reasons why that may happen i am not aware, as the login event handler is pretty plain. These changes should also catch the edge case of the player getting pearled before accepting the TOS, since it clears the old location.